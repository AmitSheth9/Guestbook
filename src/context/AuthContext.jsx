import { createContext, useState, useContext } from "react";

const authContext = createContext();

const ProvideAuth = ({ children }) => {
    const [user, setUser] = useState('');

    const login = (username, password) => {
        const loginSuccesful = username === process.env.AUTH_USERNAME && password === process.env.AUTH_PASSWORD;

        if(loginSuccesful) setUser( username );
        return loginSuccesful;
    };
    const logout = (callback) => {
        setUser(null);
        callback();
    };

    return (
        <authContext.Provider value={{ user, login, logout }}>{children}</authContext.Provider>
    );
}

const useAuth = () => {
    const context = useContext(authContext);

    if (context === undefined) {
        throw new Error(
            'Error')
    }
    return context;
}

export { ProvideAuth, useAuth }