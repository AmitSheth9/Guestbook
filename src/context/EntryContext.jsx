import { useContext, useState, createContext } from "react";

const EntryContext = createContext();

const EntryProvider = ({ children }) => {
    const [entry, setEntry] = useState([]);
    return (
        <EntryContext.Provider value={{ entry, setEntry }}>{children}</EntryContext.Provider>
    )
}

const useEntry = () => {
    const context = useContext(EntryContext)
    if(context === undefined) {
        throw new Error('Error something something')
    }
    return context
}

export { EntryProvider, useEntry }