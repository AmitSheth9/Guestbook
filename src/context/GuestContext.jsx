import { useContext, useState, createContext } from "react";

const GuestContext = createContext();

const GuestProvider = ({ children }) => {
    const [guest, setGuest] = useState({});

    return (
        <GuestContext.provider value={{user, setUser}}>{children}</GuestContext.provider>
    )
}

const useGuest = () => {
    const context = useContext(GuestContext)
    if(context === undefined) {
        throw new Error('Error')
    }
    return context
}

export { GuestProvider, useGuest };