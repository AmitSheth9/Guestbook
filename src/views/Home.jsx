import React from 'react'
import { EntryProvider } from '../context/EntryContext'
export default function Home() {
    return (
        <div>
            
        </div>
    )
}
