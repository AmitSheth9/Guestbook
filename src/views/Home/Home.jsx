import React from 'react'
import { EntryProvider } from '../../context/EntryContext'
import Header from '../Header/Header'
import Guestform from '../../components/Form/Guestform'
import DisplayEntries from '../../components/Display/DisplayEntries'
export default function Home() {
    return (
        <div>
    <EntryProvider>
        <Header/> 
        <Guestform/>
        <DisplayEntries/>
    </EntryProvider>
        </div>
    )
}
