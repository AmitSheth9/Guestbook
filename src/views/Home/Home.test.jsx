import React from 'react'
import { render, screen } from '@testing-library/react'
import Home from "./Home";
import { useGuest } from "../../context/GuestContext";
import { GuestProvider } from '../../context/GuestContext'
import { EntryProvider } from '../../context/EntryContext'
import { ProvideAuth } from '../../context/AuthContext';
it('should display Home on screen', () => {
    const { container } = render(
         
    );
    expect(container).toMatchSnapshot();

    } )