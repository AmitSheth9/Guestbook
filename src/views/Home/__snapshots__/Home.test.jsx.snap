// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display Home on screen 1`] = `
<div>
  <div>
    <div>
      <h3>
        Welcome Guest
      </h3>
    </div>
    <div>
      Sign The Guestbook!
      <form>
        <div>
          <label>
            Guest Name
            <input
              value=""
            />
          </label>
          <br />
           
        </div>
        <label>
          Guest Entry
          <input
            value=""
          />
        </label>
        <button
          type="submit"
        >
          Sign
        </button>
        <div />
      </form>
      <div />
    </div>
    <div
      class="outer-container"
    />
  </div>
</div>
`;
